src/Microsoft.TestPlatform.Extensions.HtmlLogger/HtmlLogger.cs

```diff
@@ -335,17 +335,37 @@ private void PopulateHtmlFile()
         ConsoleOutput.Instance.Information(false, htmlFilePathMessage);
     }
 
-    private string GetFilePath(string fileExtension, string fileName)
+    private string GetFilePathAndCreateFile(string fileExtension, string fileName)
     {
         var fullFileFormat = $".{fileExtension}";
-        return Path.Combine(TestResultsDirPath, string.Concat("TestResult_", fileName, fullFileFormat));
+        string fullFilePath;
+        for (short i = 0; i < short.MaxValue; i++)
+        {
+            var fileNameWithIter = i == 0 ? fileName : GetNextIterationFile(fileName, i);
+            fullFilePath = Path.Combine(TestResultsDirPath, string.Concat("TestResult_", fileNameWithIter, fullFileFormat));
```

Comment:
I would inline line 340 here

```suggestion
            fullFilePath = Path.Combine(TestResultsDirPath, $"TestResult_{fileNameWithIter}.{fileExtension}");
```